### PR TITLE
provider/azurerm: Event Hub Namespaces

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Added support for Event Hubs Namespaces:

- [x] Added a Type for Event Hub Namespaces
- [x] Website Documentation
- [x] Imported the Event Hub SDK
- [x] Run the Tests

I started / was hoping to add support for Event Hubs / Consumer Groups / Authorization Rules in the same PR - but there's bugs in [the Azure SDK for Go](https://github.com/Azure/azure-sdk-for-go/issues/413) meaning this isn't possible right now.. so I'll send a follow up when it's fixed :)

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
The tests pass.. I'm roaming so there's some latency / timeout issues - I'll re-run them when I'm on a stable connection..

```
tharvey-mbp:terraform tharvey$ TF_ACC=1 go test ./builtin/providers/azurerm -v -run TestAccAzureRMEventHubNamespace -timeout 120m
=== RUN   TestAccAzureRMEventHubNamespaceCapacity_validation
--- PASS: TestAccAzureRMEventHubNamespaceCapacity_validation (0.00s)
=== RUN   TestAccAzureRMEventHubNamespaceSku_validation
--- PASS: TestAccAzureRMEventHubNamespaceSku_validation (0.00s)
=== RUN   TestAccAzureRMEventHubNamespace_basic
--- PASS: TestAccAzureRMEventHubNamespace_basic (1051.51s)
=== RUN   TestAccAzureRMEventHubNamespace_readDefaultKeys
--- FAIL: TestAccAzureRMEventHubNamespace_readDefaultKeys (893.14s)
	testing.go:265: Step 0 error: Check failed: Check 1/5 error: Bad: Get on eventHubNamespacesClient: eventhub.NamespacesClient#Get: Failure sending request: StatusCode=0 -- Original Error: Get https://management.azure.com/subscriptions/d386467e-33de-44ff-b324-641549c30383/resourceGroups/acctestRG-4544350067092685854/providers/Microsoft.EventHub/namespaces/acctesteventhubnamespace-4544350067092685854?api-version=2015-08-01: read tcp 192.168.1.100:50289->13.87.96.212:443: read: operation timed out
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/azurerm	1944.659s

tharvey-mbp:terraform tharvey$ TF_ACC=1 go test ./builtin/providers/azurerm -v -run TestAccAzureRMEventHubNamespace_readDefaultKeys -timeout 120m
=== RUN   TestAccAzureRMEventHubNamespace_readDefaultKeys
--- PASS: TestAccAzureRMEventHubNamespace_readDefaultKeys (588.10s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	588.112s
```

One thing of note - if you create an Event Hub Namespace, and subsequently delete it - you can't create it (in a new subscription, at least) for 7 days - as noted in this screenshot. Is it worth mentioning this in the docs?

event-hubs-error

**Comment by tombuildsstuff** [comment]
```
tharvey-mbp:terraform tharvey$ TF_ACC=1 go test ./builtin/providers/azurerm -v -run TestAccAzureRMEventHubNamespace -timeout 120m
=== RUN   TestAccAzureRMEventHubNamespaceCapacity_validation
--- PASS: TestAccAzureRMEventHubNamespaceCapacity_validation (0.00s)
=== RUN   TestAccAzureRMEventHubNamespaceSku_validation
--- PASS: TestAccAzureRMEventHubNamespaceSku_validation (0.00s)
=== RUN   TestAccAzureRMEventHubNamespace_basic
--- PASS: TestAccAzureRMEventHubNamespace_basic (790.14s)
=== RUN   TestAccAzureRMEventHubNamespace_readDefaultKeys
--- PASS: TestAccAzureRMEventHubNamespace_readDefaultKeys (345.84s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	1135.997s
```

👍 all good

**Comment by tombuildsstuff** [comment]
@stack72 @jen20 I've rebased this and the previously failing Travis tests (AWS) pass fine, do I need to re-run the tests?

File: builtin/providers/azurerm/resource_arm_eventhub_namespace.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+// default connection strings and keys
+var eventHubNamespaceDefaultAuthorizationRule = "RootManageSharedAccessKey"
+
+func resourceArmEventHubNamespace() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmEventHubNamespaceCreate,
+		Read:   resourceArmEventHubNamespaceRead,
+		Update: resourceArmEventHubNamespaceCreate,
+		Delete: resourceArmEventHubNamespaceDelete,
+		Importer: &schema.ResourceImporter{
```
Comment (stack72) [inline builtin/providers/azurerm/resource_arm_eventhub_namespace.go]: Can you add a test to make sure that the import works?

File: builtin/providers/azurerm/resource_arm_eventhub_namespace.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sku": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     false,
```
Comment (stack72) [inline builtin/providers/azurerm/resource_arm_eventhub_namespace.go]: ForceNew is false by default :)

Comment (tombuildsstuff) [inline builtin/providers/azurerm/resource_arm_eventhub_namespace.go]: Fixed this.

File: builtin/providers/azurerm/resource_arm_eventhub_namespace.go (hunk 1 above)
Comment (tombuildsstuff) [inline builtin/providers/azurerm/resource_arm_eventhub_namespace.go]: Added a test but it's failing - posting this here for later as a reminder of what needs fixing:

```
TF_ACC=1 go test ./builtin/providers/azurerm -v -run TestAccAzureRMEventHubNamespace_importBasic -timeout 120m
=== RUN   TestAccAzureRMEventHubNamespace_importBasic
--- FAIL: TestAccAzureRMEventHubNamespace_importBasic (371.43s)
	testing.go:265: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
		
		(map[string]string) {
		}
		
		
		(map[string]string) (len=2) {
		 (string) (len=8) "location": (string) (len=6) "westus",
		 (string) (len=19) "resource_group_name": (string) (len=28) "acctestRG-342930116917413830"
		}
		
FAIL
exit status 1
FAIL	github.com/hashicorp/terraform/builtin/providers/azurerm	371.449s
```

**Comment by stack72** [comment]
Hi @tombuildsstuff 

This LGTM now :) Thanks for the extra work here! I am going to rebase this PR to a single commit to keep the commit log clean :)

```
% make testacc TEST=./builtin/providers/azurerm TESTARGS='-run=TestAccAzureRMEventHubNamespace_'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
2016/10/25 15:31:19 Generated command/internal_plugin_list.go
TF_ACC=1 go test ./builtin/providers/azurerm -v -run=TestAccAzureRMEventHubNamespace_ -timeout 120m
=== RUN   TestAccAzureRMEventHubNamespace_importBasic
--- PASS: TestAccAzureRMEventHubNamespace_importBasic (345.29s)
=== RUN   TestAccAzureRMEventHubNamespace_basic
--- PASS: TestAccAzureRMEventHubNamespace_basic (340.01s)
=== RUN   TestAccAzureRMEventHubNamespace_readDefaultKeys
--- PASS: TestAccAzureRMEventHubNamespace_readDefaultKeys (347.12s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/azurerm	1032.443s
```

P.

**Pull request merged**